pilot/pkg/security/authn/utils/utils.go

```diff
@@ -73,10 +83,16 @@ func BuildInboundFilterChain(mTLSMode model.MutualTLSMode, sdsUdsPath string, no
 				// include "istio", which would interfere with negotiation of the underlying
 				// protocol, e.g. HTTP/2.
 				AlpnProtocols: util.ALPNHttp,
+				// Set Minimum TLS version to match the default client version and allowed strong cipher suites for sidecars.
```

Comment:
LGTM, not sure do we care this breaks non-Envoy clients (if any) in the mesh? +@myidpt @incfly to double check.

Reply:
We have so many ALPN hacks and metadata exchange and stuff that I dont think non envoy clients (other than grpc) can even talk. and if its grpc, then we dont ahve to worry about TLS versions.